modal-go/sandbox.go

```diff
@@ -874,21 +874,37 @@ func (cps *cpStdin) Close() error {
 	return err
 }
 
+// cancelOnCloseReader is used to cancel background goroutines when the stream is closed.
+type cancelOnCloseReader struct {
+	io.ReadCloser
+	cancel context.CancelFunc
+}
+
+func (r *cancelOnCloseReader) Close() error {
+	r.cancel()
+	return r.ReadCloser.Close()
+}
+
 func outputStreamSb(cpClient pb.ModalClientClient, sandboxID string, fd pb.FileDescriptor) io.ReadCloser {
 	pr, pw := nio.Pipe(buffer.New(64 * 1024))
+	ctx, cancel := context.WithCancel(context.Background())
 	go func() {
 		defer pw.Close()
+		defer cancel()
```

Comment:
Is this defer required? At a glance it looks like the goroutine is canceling itself on exit. But if it's exiting, it does not need to cancel (it's already exiting).